zulipterminal/model.py

```diff
@@ -785,6 +786,21 @@ def get_stream_by_id(streams: List[StreamData], stream_id: int
                     sort_streams(self.pinned_streams)
                     self.controller.view.left_panel.update_stream_view()
                     self.controller.update_screen()
+        elif event['op'] == 'add':
+            def make_reduced_stream_data(stream: Any) -> StreamData:
+                # stream_id has been changed to id.
+                return StreamData({'name': stream['name'],
+                                   'id': stream['stream_id'],
+                                   'color': stream['color'],
+                                   'invite_only': stream['invite_only'],
+                                   'description': stream['description']})
+
+            subscription = event['subscriptions'][0]
+            stream_data = make_reduced_stream_data(subscription)
+            self.stream_dict[subscription['stream_id']] = stream_data
+            self.pinned_streams.append(stream_data)
```

Comment:
The new stream is not necessarily pinned.

Reply:
> @Abhirup-99 Thanks for giving this a go 
> 
> There are a few issues with the PR as it stands - how have you tested this? I tried subscribing to a stream in the webapp and the list of streams in ZT doesn't update.
> 
> I would recommend checking what the existing code does for subscriptions when they are loaded at startup, ie. any translation of the data, as well as how the stream list is updated and re-rendered in other cases.
> 
> Regarding your question in the stream, adding remove events would be good to include in the same PR.

Sorry for the embarrassing mistake, I forgot to push 1 line in the code, I was removing all print statements and additional stuff and might have deleted that as well. I am pushing this shortly along with the tests for this part.

Reply:
> The new stream is not necessarily pinned.
Fixed in the new push.